### DISCUSSION TITLE
Self hosting

### DISCUSSION THREAD

**Discussion by Justman100** [forwardemail/forwardemail.net#300] [Ideas]
A minimized version of FW, which only contains the dashboard to control the domains and aliases, as well as the mail server and its components! Without the start page, documentation or the billing system!

**Comment by Justman100** [comment]
To bring open source, of course

**Comment by shaunwarman** [comment]
Absolutely, this was originally started, but there is an ongoing effort to rewrite / optimize the MX codebase and bring that into the monorepo. From there, can focus on FE lite / self hosting.

**Comment by shaunwarman** [comment]
Progress here: https://github.com/forwardemail/forwardemail.net/pull/321

**Comment by titanism** [comment]
@shaunwarman is nearing completion of self-hosting! 🎉   this should be out soon enough I think 🤞  🚀

**Comment by titanism** [comment]
See <https://github.com/forwardemail/awesome-mail-server-providers>, a new repository we made for helping you find a provider for self-hosting your mail server.

**Reply by Justman100** [reply]
> See [forwardemail/awesome-mail-server-providers](https://github.com/forwardemail/awesome-mail-server-providers), a new repository we made for helping you find a provider for self-hosting your mail server.

What does that have to do with it now?

**Reply by shaunwarman** [reply]
Unfortunately most providers have some limitation or create some extra friction when trying to host your own email. Port 25 (smtp) blocked, no reverse pointer support (needed for email auth), bad IP reputation, etc.

**Comment by titanism** [comment]
We just merged PR #340 thanks to @shaunwarman that makes it easy to self-host Forward Email.

See the guide at <https://github.com/forwardemail/forwardemail.net/blob/master/app/views/self-hosted/index.md>

**Reply by shaunwarman** [reply]
For those that try it out, please share any feedback / concerns! We're also available in [matrix chat](https://matrix.to/#/%23forwardemail:matrix.org)
